src/coreComponents/python/modules/geosx_mesh_doctor/checks/generate_fractures.py

```diff
@@ -89,43 +89,64 @@ def build_node_to_cells(mesh: vtkUnstructuredGrid,
-        cell: vtkCell = mesh.GetCell(cell_id)
-        for face_id in faces_ids:
-            fn: Collection[int] = tuple(vtk_iter(cell.GetFace(face_id).GetPointIds()))
-            fnh = frozenset(fn)
-            if fnh not in face_nodes_hashes:
-                face_nodes_hashes.add(fnh)
-                face_nodes.append(fn)
+                        options: Options) -> FractureInfo:
+    field = options.field
+    field_values = options.field_values
+    if options.policy == "field":
+        cell_data = mesh.GetCellData()
+        if cell_data.HasArray(field):
+            f = vtk_to_numpy(cell_data.GetArray(field))
+        else:
+            raise ValueError(f"Cell field {field} does not exist in mesh, nothing done")
+        cells_to_faces: Dict[int, List[int]] = defaultdict(list)
+        # For each face of each cell, we search for the unique neighbor cell (if it exists).
+        # Then, if the 2 values of the two cells match the field requirements,
+        # we store the cell and its local face index: this is indeed part of the surface that we'll need to be split.
+        for cell_id in tqdm(range(mesh.GetNumberOfCells()), desc="Computing the cell to faces mapping"):
+            if f[cell_id] not in field_values:  # No need to consider a cell if its field value is not in the target range.
+                continue
+            cell = mesh.GetCell(cell_id)
+            for i in range(cell.GetNumberOfFaces()):
+                neighbor_cell_ids = vtkIdList()
+                mesh.GetCellNeighbors(cell_id, cell.GetFace(i).GetPointIds(), neighbor_cell_ids)
+                assert neighbor_cell_ids.GetNumberOfIds() < 2
+                for j in range(neighbor_cell_ids.GetNumberOfIds()):    # It's 0 or 1...
+                    neighbor_cell_id = neighbor_cell_ids.GetId(j)
+                    if f[neighbor_cell_id] != f[cell_id] and f[neighbor_cell_id] in field_values:
+                        cells_to_faces[cell_id].append(i)  # TODO add this (cell_is, face_id) information to the fracture_info?
+        face_nodes: List[Collection[int]] = list()
+        face_nodes_hashes: Set[FrozenSet[int]] = set()  # A temporary not to add multiple times the same face.
+        for cell_id, faces_ids in tqdm(cells_to_faces.items(), desc="Extracting the faces of the fractures"):
+            cell: vtkCell = mesh.GetCell(cell_id)
+            for face_id in faces_ids:
+                fn: Collection[int] = tuple(vtk_iter(cell.GetFace(face_id).GetPointIds()))
+                fnh = frozenset(fn)
+                if fnh not in face_nodes_hashes:
+                    face_nodes_hashes.add(fnh)
+                    face_nodes.append(fn)
+        node_to_cells: Mapping[int, Iterable[int]] = build_node_to_cells(mesh, face_nodes)
+    
+    if options.policy == "internal_surfaces":
+        node_to_cells = {}
+        face_nodes = []
+        attribute = mesh.GetCellData().GetArray(field)
```

Comment:
`attribute` is extracted for both `field` and `internal_surfaces`.
Could you use the same code (like in lines 96 -> 100)?
Of course `if attribute.GetValue(c) in field_values` line 136 becomes `if attribute[c] in field_values`.